src/subjugator/drivers/luxonis_cam/luxonis_cam/run_cam.py

```diff
@@ -0,0 +1,111 @@
+        self.stereo.setSubpixel(False)
+        self.stereo.setRectifyEdgeFillColor(0)
+        self.stereo.setInputResolution(1280, 720)
+
+        self.cam_left.out.link(self.stereo.left)
+        self.cam_right.out.link(self.stereo.right)
+
+        self.xout_color = self.__pipeline.create(dai.node.XLinkOut)
+        self.xout_color.setStreamName("cam_rgb")
+        self.cam_rgb.video.link(self.xout_color.input)
+
+        self.xout_left = self.__pipeline.create(dai.node.XLinkOut)
+        self.xout_left.setStreamName("left")
+        self.cam_left.out.link(self.xout_left.input)
+
+        self.xout_right = self.__pipeline.create(dai.node.XLinkOut)
+        self.xout_right.setStreamName("right")
+        self.cam_right.out.link(self.xout_right.input)
+
+        self.xout_depth = self.__pipeline.create(dai.node.XLinkOut)
+        self.xout_depth.setStreamName("cam_depth")
+        self.stereo.depth.link(self.xout_depth.input)
+
+        self.bridge = CvBridge()
+
+        # Launch run in a background thread so ROS can spin separately
+        threading.Thread(target=self.run, daemon=True).start()
+
+    def run(self):
+        with dai.Device(self.__pipeline) as device:
+            color_queue = device.getOutputQueue(
+                name="cam_rgb",
+                maxSize=4,
+                blocking=False,
+            )
+            left_queue = device.getOutputQueue(name="left", maxSize=4, blocking=False)
+            right_queue = device.getOutputQueue(name="right", maxSize=4, blocking=False)
+            depth_queue = device.getOutputQueue(
+                name="cam_depth",
+                maxSize=4,
+                blocking=False,
+            )
+
+            while rclpy.ok():
+                color_frame = (
+                    color_queue.get().getCvFrame() if color_queue.has() else None
+                )
+                left_frame = left_queue.get().getCvFrame() if left_queue.has() else None
+                right_frame = (
+                    right_queue.get().getCvFrame() if right_queue.has() else None
+                )
+                depth_frame = (
+                    depth_queue.get().getCvFrame() if depth_queue.has() else None
+                )
+
+                # print(color_frame is not None, mono_frame is not None, depth_frame is not None)
+
+                if color_frame is not None:
+                    msg = self.bridge.cv2_to_imgmsg(color_frame, "bgr8")
+                    self.rgb_pub.publish(msg)
+                if left_frame is not None:
+                    msg = self.bridge.cv2_to_imgmsg(left_frame, "mono8")
+                    self.left_pub.publish(msg)
+                if right_frame is not None:
+                    msg = self.bridge.cv2_to_imgmsg(right_frame, "mono8")
+                    self.right_pub.publish(msg)
+                if depth_frame is not None:
+                    msg = self.bridge.cv2_to_imgmsg(depth_frame, "mono16")
+                    self.depth_pub.publish(msg)
+
```

Comment:
Not the biggest deal in the world, but this code is a tad bit repetitive. You could move this code into a function that just takes the image format, publisher object, etc., but it's no big deal